transformers/modeling_bert.py

```diff
@@ -277,9 +340,49 @@ def prune_heads(self, heads):
         self.self.all_head_size = self.self.attention_head_size * self.self.num_attention_heads
         self.pruned_heads = self.pruned_heads.union(heads)
 
-    def forward(self, input_tensor, attention_mask=None, head_mask=None):
-        self_outputs = self.self(input_tensor, attention_mask, head_mask)
-        attention_output = self.output(self_outputs[0], input_tensor)
+    def forward(self, hidden_states, attention_mask=None, head_mask=None):
+        self_outputs = self.self(hidden_states, attention_mask, head_mask)
+        attention_output = self.output(self_outputs[0], hidden_states)
+        outputs = (attention_output,) + self_outputs[1:]  # add attentions if we output them
+        return outputs
+
+
+class BertDecoderAttention(nn.Module):
```

Comment:
Same here, can we modify `BertAttention` so that it stays fully backward compatible but has an additional flag and input in the forward pass to be able to use it for encoder-decoder attention as well?